pkg/chunked/cache_linux.go

```diff
@@ -455,19 +529,18 @@ func prepareMetadata(manifest []byte, format graphdriver.DifferOutputFormat) ([]
 	return r, nil
 }
 
-func (c *layersCache) addLayer(id string, cacheFile *cacheFile) error {
+func (c *layersCache) createLayer(id string, cacheFile *cacheFile, mmapBuffer []byte) (*layer, error) {
 	target, err := c.store.DifferTarget(id)
 	if err != nil {
-		return fmt.Errorf("get checkout directory layer %q: %w", id, err)
+		return nil, fmt.Errorf("get checkout directory layer %q: %w", id, err)
 	}
-
 	l := layer{
-		id:        id,
-		cacheFile: cacheFile,
-		target:    target,
+		id:         id,
+		cacheFile:  cacheFile,
+		target:     target,
+		mmapBuffer: mmapBuffer,
 	}
```

Comment:
In here, we can do something like
```go
if mmapBuffer != nil {
    runtime.SetFinalizer(l, l.release)
}
```

(and amend release() to do `SetFinalizer(l, nil)`).

Reply:
changed to use `runtime.SetFinalizer`